core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -758,6 +758,8 @@ public PlanOptimizers(
                         ImmutableSet.of(
                                 new ApplyPreferredTableWriterPartitioning(),
                                 new ApplyPreferredTableExecutePartitioning())),
+                // Make sure to run ReplicateJoinAndSemiJoinInDelete before ReorderJoins and AddExchanges
+                new ReplicateJoinAndSemiJoinInDelete(),
```

Comment:
I don't understand why ReplicateSemiJoinInDelete/ReplicateJoinAndSemiJoinInDelete needs to be run before ReorderJoins.
I think ReplicateSemiJoinInDelete/ReplicateJoinAndSemiJoinInDelete would undo distirbution type picked by ReorderJoins. What am i missing?

Reply:
Since ReorderJoins can flip the JoinNode - we would like to avoid it for the PlanNode between Delete and TableScan. So we run them before applying these optimizer.